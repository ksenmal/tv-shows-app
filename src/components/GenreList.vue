<script setup>
import { RouterLink } from "vue-router";
import { genreList } from "../utils/constants";
</script>

<template>
  <nav>
    <RouterLink
      v-for="genre in genreList"
      :to="`/${genre.route}`"
      :key="genre.name"
      >{{ genre.name }}</RouterLink
    >
  </nav>
</template>

<style>
nav {
  width: 100%;
  font-size: 1rem;
  text-align: center;
  margin-bottom: 1rem;
}

nav a.router-link-exact-active {
  color: var(--color-text);
}

nav a.router-link-exact-active:hover {
  background-color: transparent;
}

nav a {
  display: inline-block;
  padding: 0 1rem;
  background-color: var(--color-background-grey);
  border-radius: 5px;
  margin: 2px;
}

@media (min-width: 1024px) {
  nav {
    font-size: 1rem;
    padding: 1rem 0;
  }
}
</style>
