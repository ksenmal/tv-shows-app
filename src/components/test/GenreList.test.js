import { shallowMount } from "@vue/test-utils";
import GenreList from "../GenreList.vue";
import { describe, expect, test } from "vitest";

describe("GenreList suite", () => {
  test("Snapshot", () => {
    const wrapper = shallowMount(GenreList);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
