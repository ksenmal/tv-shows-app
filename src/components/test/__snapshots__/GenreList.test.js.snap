// Vitest Snapshot v1

exports[`GenreList suite > Snapshot 1`] = `
"<nav>
  <router-link-stub to=\\"/action\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/adventure\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/anime\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/children\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/comedy\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/crime\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/drama\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/family\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/fantasy\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/horror\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/medical\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/music\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/nature\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/sports\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/supernatural\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/thriller\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/travel\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
  <router-link-stub to=\\"/western\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
</nav>"
`;
